src/arithmetic/algebraic_expression.c

```diff
@@ -182,21 +182,21 @@ inline void _AlgebraicExpression_Execute_MUL(GrB_Matrix C, GrB_Matrix A, GrB_Mat
  * operation (multiply) and a number of operands. */
 AlgebraicExpressionResult *AlgebraicExpression_Execute(AlgebraicExpression *ae) {
     assert(ae);
-    
-    /* TODO: there are cases where we don't need to transpose,
-     * for example when there's only a single operand and both the operand 
-     * is required to be transposed and the result is requested to be
-     * transposed, in that case transpose(transpose(A)) = A. 
-     * And so in that case we'll perform two unnecessary transpose. */
 
     GrB_Matrix A = ae->operands[0].operand;
     GrB_Matrix C = A;
     int operand_count = ae->operand_count;
-    
+
     if(operand_count == 1) {
+        /* AlgebraicExpression_Transpose guarantees that
+         * either the expression is marked as transposed or
+         * the single term, but never both. */
+        assert(!(ae->transpose && ae->operands[0].transpose));
+
+        /* When transposing we must duplicate, as graph matrices
+         * should be considered immutable. */
         if(ae->operands[0].transpose) {
             GrB_Matrix dup;
-            /* TODO: replace assert with if, log, exit 1. */
             assert(GrB_Matrix_dup(&dup, A) == GrB_SUCCESS);
             assert(GrB_transpose(dup, NULL, NULL, dup, NULL) == GrB_SUCCESS);
```

Comment:
As we discussed, we might be able to replace the `GrB_Matrix_dup` with the creation of an empty matrix to use as the first argument to `GrB_transpose`.